include/common/environment_variables.h

```diff
@@ -0,0 +1,71 @@
+#ifndef COMMON_ENVIRONMENT_VARIABLES
+#define COMMON_ENVIRONMENT_VARIABLES
+
+#include "mpmc_queue.h"
+
+#include <sstream>
+#include <thread>
+#include <cstdlib>
+#include <cstring>
+#include <string>
+
+namespace grppi
+{
+
+namespace{
```

Comment:
Unnamed namespace in headers goes against secure C++ coding guidelines:

https://wiki.sei.cmu.edu/confluence/display/cplusplus/DCL59-CPP.+Do+not+define+an+unnamed+namespace+in+a+header+file

This is also CppCore guidelines SF.21

https://wiki.sei.cmu.edu/confluence/display/cplusplus/DCL59-CPP.+Do+not+define+an+unnamed+namespace+in+a+header+file

Reply:
Design alternative.

When you want to get the environment you want to get all values. You can have a class whose default constructor populates values from environment variables.

For example `execution_parameters`

Reply:
I have defined the class "environment_variables" instead of the namespace.